### PR TITLE
Added note when no posts are available/highlighted 'Create a Post' button

### LINKED ISSUE #1367: [Desktop][Mobile] Need to display a general error message if there are no relevant posts available.
As a user,
When I filter the posts as per "Location", "Type" and "Providers" and if no relevant posts exists as per my filters,
Then I should be displayed the following message: "Sorry, there are currently no relevant posts available. Please try using different tags or create your own post."

image

**NOTE:** This error message should be displayed irrespective of whether the user is logged in or not.

### PR DESCRIPTION
**Pull request by kterry1**
# What does this pull request aim to achieve? 💯
- When no posts are available there is now a note letting the user know there are no posts available and to change the filters or create a post.
- Also, the 'Create a Post' button is highlighted so the user knows where to click when the feed is empty.

Here is a demonstration of it in action: https://share.getcloudapp.com/8LuPp4Q7 

You can use the following filter in the Build URL to test: https://share.getcloudapp.com/xQuLzlvB

_Please be concise and link any related issue(s):_
Resolves: #1367 

- [X] I have checked that no one else is working on similar changes.
- [X] I have read the latest [**README**](README.md) and understand the development workflow.
- [X] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix. The branch is created in a cloned version of this repo, **not a forked repo**.
- [X] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [X] There are no merge conflicts.
- [X] There are no console warnings and errors.
- [X] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".

### CODE REVIEW

**Comment by kterry1** [comment]
I tag you both Josue and Josh a lot. Let me know if there is someone else you recommend I tag for review so I'm not bugging you both all the time.

File: client/src/pages/Feed.js [hunk 1]
```diff
@@ -221,6 +229,23 @@ const HeaderWrapper = styled.div`
...
+`
+
+const buttonPulse = styled.button`
+  background-color: rgba(255,255,0, .4);
+  padding: 0 .2em;
+  border-radius: 3em;
+  height: 4em;
+  position: relative;
+  top: 1.3em;
+`
```
Comment (idjevm) [inline client/src/pages/Feed.js:247]: add `;`

Comment (idjevm) [inline client/src/pages/Feed.js]: add `;`

File: client/src/pages/Feed.js [hunk 2]
```diff
@@ -221,6 +229,23 @@ const HeaderWrapper = styled.div`
   }
 `;
 
+const NoPosts = styled.div`
+    text-align: center;
+    position: relative;
+    top: 2em;
+    color: ${theme.colors.orangeRed};
+    font-size: 1.2em;
+`
```
Comment (idjevm) [inline client/src/pages/Feed.js]: add `;`

File: client/src/pages/Feed.js [hunk 3]
```diff
@@ -562,6 +587,7 @@ const Feed = (props) => {
     };
   }, [scrollObserver, bottomBoundaryRef]);
 
+
```
Comment (idjevm) [inline client/src/pages/Feed.js]: Remove extra line

File: client/src/pages/Feed.js [hunk 4]
```diff
@@ -186,6 +186,7 @@ const LayoutWrapper = styled(Layout)`
   }
 `;
 
+
```
Comment (idjevm) [inline client/src/pages/Feed.js]: remove extra line

File: client/src/pages/Feed.js [hunk 5]
```diff
@@ -202,14 +203,21 @@ const HeaderWrapper = styled.div`
   }
   button {
     align-items: center;
-    background-color: transparent;
+    background-color: ${({empty}) => empty ? "rgba(255,255,0, .3)" :
+    "transparent"
+    };
```
Comment (idjevm) [inline client/src/pages/Feed.js]: Check your prettier settings, the formatting seems odd.

File: client/src/pages/Feed.js [hunk 6]
```diff
@@ -221,6 +229,23 @@ const HeaderWrapper = styled.div`
   }
 `;
 
+const NoPosts = styled.div`
+    text-align: center;
+    position: relative;
+    top: 2em;
+    color: ${theme.colors.orangeRed};
```
Comment (idjevm) [inline client/src/pages/Feed.js]: Formatting seems a bit diff here. Perhaps check prettier?

File: client/src/pages/Feed.js [hunk 7]
```diff
@@ -677,12 +707,14 @@ const Feed = (props) => {
               <ErrorAlert message={postsError.message} />
             )}
             {isLoading ? <Loader /> : <></>}
+            {emptyFeed() ? <NoPosts>Sorry, there are currently no relevant posts available. Please try using a different filter search or create a post.</NoPosts>
```
Comment (idjevm) [inline client/src/pages/Feed.js]: Please check your prettier settings.

**Review by idjevm — CHANGES_REQUESTED** [review]
Mmmh, I am not sure about this one though, I think we can ask the design team for a design of when there are not posts available... This should be an ok solution meanwhile.

**Comment by kterry1** [comment]
Thanks @idjevm. It looks like Prettier wasn't formatting the doc. I made the fix but before I push the new commit up, can we get a design team member to provide another highlight recommendation? @dhruvdpatel are you able to chime in here since you created #1420.

**Comment by kterry1** [comment]
@dhruvdpatel Clarified what he wanted here. I'll make the changes and push it up later today.

**Comment by kterry1** [comment]
Here is the new look: https://share.getcloudapp.com/04uqLWvD

I'll push up the code in a bit.

**Review by nadinejuraschek — APPROVED** [review]
lgtm!

**Review by idjevm — APPROVED** [review]
Codewise and logic wise, this makes sense. Could anyone from design check this though? @Yu-Alexa-Zhao , is there a chance your team could take a look at this? We will merge this for now but let's please create a design for this one to make it more nicer looking, etc.

**Comment by idjevm** [comment]
Will merge this now, any new implementations, please create a new issue, and ask for the design there @dhruvdpatel @tejasm2 @Yu-Alexa-Zhao

**Comment by yu-zhao-xyz** [comment]
Got it, I will ask the design team to create a better looking no result screen here

**Pull request merged**